crates/swift-bridge-ir/src/parse/parse_enum/enum_attributes.rs

```diff
@@ -0,0 +1,104 @@
+
+#[derive(Default)]
+pub(super) struct SharedEnumAllAttributes {
+    pub(super) swift_bridge: SharedEnumSwiftBridgeAttributes,
+    pub(super) derive: DeriveAttrs,
+}
+
+impl SharedEnumAllAttributes {
+    pub(super) fn from_attributes(attribs: &[Attribute]) -> Result<Self, syn::Error> {
+        let mut attributes = SharedEnumAllAttributes::default();
+
+        for attr in attribs {
+            let attribute_name = attr.path.to_token_stream().to_string();
+
+            match attribute_name.as_str() {
+                "derive" => {
+                    let parser =
+                        syn::punctuated::Punctuated::<Path, syn::Token![,]>::parse_terminated;
+                    let args = attr.parse_args_with(parser)?;
+                    for arg in args.into_iter() {
+                        match arg.get_ident().unwrap().to_string().as_str() {
+                            "Debug" => attributes.derive.debug = true,
+                            _ => todo!("Unsupported derive macro; please see https://github.com/chinedufn/swift-bridge/issues/190#issuecomment-1463234027"),
+                        }
+                    }
+                }
+                "swift_bridge" => attributes.swift_bridge = attr.parse_args()?,
+                _ => todo!("Push unsupported attribute error."),
+            };
+        }
+
+        Ok(attributes)
+    }
+}
+
+pub(super) enum EnumAttr {
+    AlreadyDeclared,
+    Error(EnumAttrParseError),
+    SwiftName(LitStr),
+}
+
+pub(super) enum EnumAttrParseError {
+    UnrecognizedAttribute(Ident),
+}
+
+#[derive(Default)]
+pub(super) struct SharedEnumSwiftBridgeAttributes {
+    pub already_declared: bool,
+    pub swift_name: Option<LitStr>,
+}
+
+impl SharedEnumSwiftBridgeAttributes {
+    pub(super) fn store_attrib(&mut self, attrib: EnumAttr) -> syn::Result<()> {
+        match attrib {
+            EnumAttr::AlreadyDeclared => self.already_declared = true,
+            EnumAttr::Error(_err) => {} // TODO: error handling
+            EnumAttr::SwiftName(name) => self.swift_name = Some(name),
+        };
+        Ok(())
+    }
+}
+
+impl Parse for SharedEnumSwiftBridgeAttributes {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut attributes = SharedEnumSwiftBridgeAttributes::default();
+
+        let punctuated =
+            syn::punctuated::Punctuated::<EnumAttr, syn::Token![,]>::parse_terminated(input)?;
+
+        for attr in punctuated.into_iter() {
+            attributes.store_attrib(attr)?;
+        }
+
+        Ok(attributes)
+    }
+}
+
+impl Parse for EnumAttr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let key: Ident = input.parse()?;
+
+        let attr = match key.to_string().as_str() {
+            "already_declared" => EnumAttr::AlreadyDeclared,
+            "swift_name" => {
+                input.parse::<Token![=]>()?;
+
+                let name = input.parse()?;
+                EnumAttr::SwiftName(name)
+            }
+            _ => {
```

Comment:
Let's add a UI test for this:

We can first combine the existing unrecognized attribute tests into a single `unrecognized-attribute.rs` and `unrecognized-attribute.stderr` UI test instead of having 3 different files.

https://github.com/chinedufn/swift-bridge/blob/3c0d00da33fdf19e201eb121c7e6e370897f8070/crates/swift-bridge-macro/tests/ui/unrecognized-argument-attribute.rs

https://github.com/chinedufn/swift-bridge/blob/3c0d00da33fdf19e201eb121c7e6e370897f8070/crates/swift-bridge-macro/tests/ui/unrecognized-function-attribute.rs

https://github.com/chinedufn/swift-bridge/blob/3c0d00da33fdf19e201eb121c7e6e370897f8070/crates/swift-bridge-macro/tests/ui/unrecognized-opaque-type-attribute.rs

---

Then we can add an enum to the unrecognized attribute UI test file

---

Then we can get the test passing using something like:

```rust
#[derive(Default)]
pub(super) struct SharedEnumSwiftBridgeAttributes {
    pub already_declared: bool,
    pub swift_name: Option<LitStr>,
    pub errors: Vec<ParseError>
}

impl SharedEnumSwiftBridgeAttributes {
    pub(super) fn store_attrib(&mut self, attrib: EnumAttr) -> syn::Result<()> {
        match attrib {
            EnumAttr::AlreadyDeclared => self.already_declared = true,
            EnumAttr::Error(err) => { self.errors.push(error) },
            EnumAttr::SwiftName(name) => self.swift_name = Some(name),
        };
        Ok(())
    }
}
```

Then in https://github.com/chinedufn/swift-bridge/blob/b23f7d3416b740d0bd920c6edce8e42d8dfe3374/crates/swift-bridge-ir/src/parse/parse_enum.rs#L16-L42

We can do something like:

```rust
for err in attribs.errors {
    self.errors.push(err);
}
```